units/generate_tasks.go

```diff
@@ -167,6 +173,22 @@ func (j *generateTasksJob) Run(ctx context.Context) {
 	}))
 }
 
+func parseProjectsAsString(jsonStrings []string) ([]model.GeneratedProject, error) {
+	catcher := grip.NewBasicCatcher()
+	var projects []model.GeneratedProject
+	for _, f := range jsonStrings {
+		p, err := model.ParseProjectFromJSONString(f)
+		if err != nil {
+			catcher.Add(err)
+		}
+		projects = append(projects, p)
+	}
+	if catcher.HasErrors() {
+		return nil, catcher.Resolve()
```

Comment:
I think this catcher.Resolve() should be wrapped to easily distinguish an error here from an error in parserProjects.

Also, if we don't mind returning a non-nil object with an error (which we seem to have done before), you can just return `projects, catcher.Resolve()` without checking the error, since no other logic happens after that.

Reply:
The wrapping is by the caller, which I think is the right place, and does what you're looking for.

Did the second thing.